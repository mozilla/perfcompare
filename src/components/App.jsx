import SearchView from './SearchView'

function App() {
    return <SearchView />
}

export default App
