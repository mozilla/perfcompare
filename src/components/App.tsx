import Alert from '@mui/material/Alert';
import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';
import Link from '@mui/material/Link';
import { ThemeProvider } from '@mui/material/styles';
import { SnackbarProvider } from 'notistack';
import { HashRouter as Router, Route, Routes } from 'react-router-dom';

import useProtocolTheme from '../theme/protocolTheme';
import CompareResultsView from './CompareResults/CompareResultsView';
import SearchView from './Search/SearchView';
import FeedbackAlert from './Shared/FeedbackAlert';
import SnackbarCloseButton from './Shared/SnackbarCloseButton';
import ToggleDarkMode from './Shared/ToggleDarkModeButton';

function App() {
  const { mode, toggleColorMode, protocolTheme } = useProtocolTheme();
  return (
    <ThemeProvider theme={protocolTheme}>
      <SnackbarProvider
        maxSnack={3}
        autoHideDuration={6000}
        action={(snackbarKey) => (
          <SnackbarCloseButton snackbarKey={snackbarKey} />
        )}
      >
        <CssBaseline />
        <Alert severity="warning" sx={{ textAlign: 'center' }}>
          This is an unstable <strong>pre-release</strong> version. Some
          features may not yet be supported. Please file any bugs on the{' '}
          <Link color="secondary" href="https://github.com/mozilla/perfcompare/issues">
            Github Repo
          </Link>
          .
        </Alert>
        <Box display="flex" justifyContent="flex-end" alignItems="flex-end">
          <FeedbackAlert />
          <ToggleDarkMode
            toggleColorMode={toggleColorMode}
            theme={protocolTheme}
          />
        </Box>
        <Router>
          <Routes>
            <Route path="/" element={<SearchView />} />
            <Route
              path="/compare-results"
              element={<CompareResultsView mode={mode} />}
            />
          </Routes>
        </Router>
      </SnackbarProvider>
    </ThemeProvider>
  );
}

export default App;
