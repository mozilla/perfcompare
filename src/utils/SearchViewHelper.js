import store from '../common/store';
import {
  updateSearchValue,
  updateSearchResults,
  updateRepository,
} from '../reducers/searchSlice';
import {
  fetchRecentRevisions,
  fetchRevisionByID,
  fetchRevisionsByAuthor,
} from '../thunks/searchThunk';

const SearchViewHelper = {
  searchByRevisionOrEmail(repository, search) {
    const emailMatch = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    const longHashMatch = /[a-zA-Z0-9]{40}/;

    if (emailMatch.test(search)) {
      store.dispatch(fetchRevisionsByAuthor({ repository, search }));
    } else if (longHashMatch.test(search)) {
      store.dispatch(fetchRevisionByID({ repository, search }));
    }
  },

  handleChangeDropdown(event) {
    const repository = event.target.innerText;
    const search = store.getState().search.searchValue;

    // Update state with selected repository
    store.dispatch(updateRepository(repository));
    // If repository is selected after search value is input, fetch search results
    if (search !== '') {
      SearchViewHelper.searchByRevisionOrEmail(repository, search);
    }
    // Fetch 10 most recent revisions when repository changes
    store.dispatch(fetchRecentRevisions(repository));
  },

  handleChangeSearch(event) {
    const search = event.target.value;
    const { repository } = store.getState().search;
    store.dispatch(updateSearchValue(search));

    // If search input is cleared, clear results
    if (search === '') store.dispatch(updateSearchResults([]));

    if (repository === '') {
      console.log('Please select a repository');
    } else {
      SearchViewHelper.searchByRevisionOrEmail(repository, search);
    }
  },
};

export default SearchViewHelper;
