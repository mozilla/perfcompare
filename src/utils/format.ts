const dateFormatter = new Intl.DateTimeFormat('en-US', {
  dateStyle: 'medium',
});

export function formatDate(date: Date) {
  return dateFormatter.format(date);
}

export function formatDateRange(date1: Date, date2: Date) {
  return formatDate(date1) + ' - ' + formatDate(date2);
}
