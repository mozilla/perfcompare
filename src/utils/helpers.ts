import {
  frameworkMap,
  devToolsFramework,
  baseDocsURL,
  removedOldTestDevTools,
  nonDocumentedTestsDevTools,
  supportedPerfdocsFrameworks,
} from '../common/constants';
import type { Repository, Changeset } from '../types/state';
import type { Framework, SupportedPerfdocsFramework } from '../types/types';

const truncateHash = (revision: Changeset['revision']) => revision.slice(0, 12);

// return only most recent commit message
// first commit is usually 'try_task_config'
const getLatestCommitMessage = (item: Changeset) => {
  const { repository_id: repositoryId, revisions } = item;
  const isTry = repositoryId === 4;
  const lastUsefulRevision =
    isTry && revisions.length > 1 ? revisions[1] : revisions[0];
  const lastUsefulSummary = lastUsefulRevision.comments.slice(
    0,
    lastUsefulRevision.comments.indexOf('\n'),
  );
  return lastUsefulSummary;
};

// ensure all numbers display two digits
const formatNumber = (number: number): string => {
  if (number < 10) {
    return `0${number}`;
  } else return String(number);
};

const formatDate = (timestamp: number): string => {
  const date = new Date(timestamp * 1000);
  const day = date.getDate();
  const month = date.getMonth() + 1;
  const year = String(date.getFullYear()).slice(2);
  const hours = date.getHours();
  const minutes = date.getMinutes();

  return `${formatNumber(month)}/${formatNumber(day)}/${year} ${formatNumber(
    hours,
  )}:${formatNumber(minutes)}`;
};

const getTreeherderURL = (
  revision: Changeset['revision'],
  repository: Repository['name'],
) =>
  `https://treeherder.mozilla.org/jobs?repo=${repository}&revision=${revision}`;

const createDevtoolsDocsUrl = (
  supportedFramework: string,
  urlReadySuite: string,
  suite: string,
) => {
  let linkSupported = true;

  let devtoolsDocsURL = `${baseDocsURL}/devtools/tests/${supportedFramework}.html#${urlReadySuite}`;
  if (
    suite === removedOldTestDevTools ||
    nonDocumentedTestsDevTools.includes(suite)
  ) {
    devtoolsDocsURL = '';
    linkSupported = false;
  }
  return { devtoolsDocsURL, linkSupported };
};

const getDocsURL = (suite: string, framework_id: Framework['id']) => {
  const framework = frameworkMap[framework_id];
  const supportedFramework =
    supportedPerfdocsFrameworks[framework as SupportedPerfdocsFramework];
  const urlReadySuite = suite.replace(/:|\s|\.|_/g, '-').toLowerCase();

  let docsURL = '';
  let isLinkSupported = true;

  if (framework_id === devToolsFramework.id) {
    const { devtoolsDocsURL, linkSupported } = createDevtoolsDocsUrl(
      supportedFramework,
      urlReadySuite,
      suite,
    );

    isLinkSupported = linkSupported;
    docsURL = devtoolsDocsURL;
  } else if (supportedFramework) {
    docsURL = `${baseDocsURL}/testing/perfdocs/${supportedFramework}.html#${urlReadySuite}`;
  } else {
    isLinkSupported = false;
  }
  return { docsURL, isLinkSupported };
};

// TO DO: Review if this method is still needed
const setConfidenceClassName = (confidenceText: string | null) => {
  return confidenceText || 'unknown-confidence';
};

const swapArrayElements = <T>(
  array: T[],
  index1: number,
  index2: number,
): T[] => {
  if (
    index1 !== index2 &&
    [index1, index2].every((index) => index < array.length && index >= 0)
  ) {
    const newArray = [...array];
    [newArray[index1], newArray[index2]] = [newArray[index2], newArray[index1]];
    return newArray;
  }
  return array;
};

export {
  formatDate,
  getLatestCommitMessage,
  getTreeherderURL,
  setConfidenceClassName,
  swapArrayElements,
  truncateHash,
  getDocsURL,
};
