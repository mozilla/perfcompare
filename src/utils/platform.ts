import type { PlatformShortName } from '../types/state';

export const getPlatformShortName = (
  platformName: string,
): PlatformShortName => {
  if (platformName.toLowerCase().includes('linux')) return 'Linux';
  if (
    platformName.toLowerCase().includes('osx') ||
    platformName.toLowerCase().includes('os x')
  )
    return 'OSX';
  if (platformName.toLowerCase().includes('windows')) return 'Windows';
  if (platformName.toLowerCase().includes('android')) return 'Android';
  return 'Unspecified';
};
