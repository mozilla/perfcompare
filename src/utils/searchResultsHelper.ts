import type { Revision } from '../types/state';

const truncateHash = (revision: Revision['revision']) => revision.slice(0, 12);

// return only most recent commit message
// first commit is usually 'try_task_config'
const getLatestCommitMessage = (item: Revision) =>
  item.revisions[item.revisions.length - 1].comments;

export { truncateHash, getLatestCommitMessage };
