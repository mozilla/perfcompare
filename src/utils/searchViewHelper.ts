import type { Repository, Revision } from '../types/state';

const truncateHash = (revision: Revision['revision']) => revision.slice(0, 12);

// return only most recent commit message
// first commit is usually 'try_task_config'
const getLatestCommitMessage = (item: Revision) =>
  item.revisions[item.revisions.length - 1].comments;

// ensure all numbers display two digits
const formatNumber = (number: number): string => {
  if (number < 10) {
    return `0${number}`;
  } else return String(number);
};

const formatDate = (timestamp: number): string => {
  const date = new Date(timestamp * 1000);
  const day = date.getDate();
  const month = date.getMonth() + 1;
  const year = String(date.getFullYear()).slice(2);
  const hours = date.getHours();
  const minutes = date.getMinutes();

  return `${formatNumber(month)}/${formatNumber(day)}/${year} ${formatNumber(
    hours,
  )}:${formatNumber(minutes)}`;
};

const getTreeherderURL = (
  revision: Revision['revision'],
  repository: Repository['name'],
) =>
  `https://treeherder.mozilla.org/jobs?repo=${repository}&revision=${revision}`;

export { formatDate, getLatestCommitMessage, getTreeherderURL, truncateHash };
