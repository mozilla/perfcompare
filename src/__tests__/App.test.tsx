import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';

import App from '../components/App';
import { render } from './utils/setupTests';
import { screen } from './utils/test-utils';

describe('App', () => {
  test('Should render search view on default route', async () => {
    render(<App />);

    // Title appears
    expect(screen.getByText(/PerfCompare/i)).toBeInTheDocument();

    await act(async () => void jest.runOnlyPendingTimers());
    const homeText = screen.getByText('Compare with a base or over time');

    expect(homeText).toBeInTheDocument();
  });

  test('Should switch between dark mode and light mode on toggle', async () => {
    // set delay to null to prevent test time-out due to useFakeTimers
    const user = userEvent.setup({ delay: null });

    render(<App />);

    const darkModeButton = screen.getByLabelText('Dark mode');

    await user.click(darkModeButton);
    expect(screen.queryByLabelText('Light mode')).toBeInTheDocument();

    await user.click(darkModeButton);
    expect(screen.queryByLabelText('Light mode')).not.toBeInTheDocument();
  });
});
