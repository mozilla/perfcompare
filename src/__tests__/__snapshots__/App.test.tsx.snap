// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App CompareResults or CompareOverTime loader Should render an error page for compare over time when the treeherder request fails with an error 400 1`] = `
<body>
  <div>
    <div
      aria-live="assertive"
      class="alert-container"
      role="alert"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard fld8l4l css-1qrenvf-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-2fbsan-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        <div
          class="banner-text"
        >
          This is a recently-released version, and we are actively addressing bugs and enhancing performance. If you experience any issues, please
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
            href="https://bugzilla.mozilla.org/enter_bug.cgi?product=Testing&component=PerfCompare&status_whiteboard=[pcf]"
            target="_blank"
          >
            report it on Bugzilla
          </a>
          . 
          Find us at
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
            href="https://matrix.to/#/#perfcompare:mozilla.org"
            target="_blank"
          >
            #perfcompare:mozilla.org on Matrix
          </a>
          .
        </div>
      </div>
    </div>
    <div
      class="fcndgf6"
    >
      <div
        class="MuiGrid-root header-container container_f1ocj467 css-plh9qo-MuiGrid-root"
      >
        <div
          class="toggle-dark-mode f10ie5uy MuiBox-root css-0"
        >
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart toggle-text css-z8chs3-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium toggle-switch toggle-dark-mode css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1n50hym-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    aria-label="Dark mode switch"
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    id="Dark mode"
                    name="toggle-dark-mode"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13qdlub-MuiTypography-root"
              >
                Dark mode
              </span>
            </label>
          </div>
        </div>
        <div
          class="header-text MuiBox-root css-0"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter MuiTypography-gutterBottom perfcompare-header css-160ox9e-MuiTypography-root"
          >
            PerfCompare
          </h1>
        </div>
      </div>
      <section
        class="container_fk3n580"
      >
        <a
          aria-label="link to home"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
          href="/"
        >
          <div
            class="css-1d9cypr-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mecf1h-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <p>
              Home
            </p>
          </div>
        </a>
        <p>
          Error: 
          Error when requesting treeherder: Treeherder request error
        </p>
        <p>
          More information about this error has been written to the Web Console.
        </p>
      </section>
    </div>
  </div>
</body>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 1`] = `
<p>
  Error: 
  The parameter baseRev is missing.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 2`] = `
<p>
  Error: 
  The parameter baseRepo is missing.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 3`] = `
<p>
  Error: 
  The parameter baseRepo is missing.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 4`] = `
<p>
  Error: 
  There should be as many "newRepo" parameters as there are "newRev" parameters.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 5`] = `
<p>
  Error: 
  There should be as many "newRepo" parameters as there are "newRev" parameters.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 6`] = `
<p>
  Error: 
  The parameter baseRepo "UNKNOWN" should be one of mozilla-central, try, mozilla-beta, mozilla-release, autoland, fenix.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 7`] = `
<p>
  Error: 
  Every parameter newRepo "UNKNOWN" should be one of mozilla-central, try, mozilla-beta, mozilla-release, autoland, fenix.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 8`] = `
<p>
  Error: 
  Every parameter newRepo "UNKNOWN" should be one of mozilla-central, try, mozilla-beta, mozilla-release, autoland, fenix.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 9`] = `
<p>
  Error: 
  The parameter interval is missing.
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 10`] = `
<p>
  Error: 
  The parameter framework should be a number, but it is "FOO".
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 11`] = `
<p>
  Error: 
  The parameter framework should be a number, but it is "FOO".
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 12`] = `
<p>
  Error: 
  The parameter framework isn't a valid value: "25".
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 13`] = `
<p>
  Error: 
  The parameter framework isn't a valid value: "25".
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 14`] = `
<p>
  Error: 
  The parameter interval should be a number, but it is "FOO".
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the requested URL is invalid 15`] = `
<p>
  Error: 
  The parameter interval isn't a valid value: "1000".
</p>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the treeherder request fails with an error 400 1`] = `
<body>
  <div>
    <div
      aria-live="assertive"
      class="alert-container"
      role="alert"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard fld8l4l css-1qrenvf-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-2fbsan-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        <div
          class="banner-text"
        >
          This is a recently-released version, and we are actively addressing bugs and enhancing performance. If you experience any issues, please
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
            href="https://bugzilla.mozilla.org/enter_bug.cgi?product=Testing&component=PerfCompare&status_whiteboard=[pcf]"
            target="_blank"
          >
            report it on Bugzilla
          </a>
          . 
          Find us at
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
            href="https://matrix.to/#/#perfcompare:mozilla.org"
            target="_blank"
          >
            #perfcompare:mozilla.org on Matrix
          </a>
          .
        </div>
      </div>
    </div>
    <div
      class="fcndgf6"
    >
      <div
        class="MuiGrid-root header-container container_f1ocj467 css-plh9qo-MuiGrid-root"
      >
        <div
          class="toggle-dark-mode f10ie5uy MuiBox-root css-0"
        >
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart toggle-text css-z8chs3-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium toggle-switch toggle-dark-mode css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1n50hym-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    aria-label="Dark mode switch"
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    id="Dark mode"
                    name="toggle-dark-mode"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13qdlub-MuiTypography-root"
              >
                Dark mode
              </span>
            </label>
          </div>
        </div>
        <div
          class="header-text MuiBox-root css-0"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter MuiTypography-gutterBottom perfcompare-header css-160ox9e-MuiTypography-root"
          >
            PerfCompare
          </h1>
        </div>
      </div>
      <section
        class="container_fk3n580"
      >
        <a
          aria-label="link to home"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
          href="/"
        >
          <div
            class="css-1d9cypr-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mecf1h-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <p>
              Home
            </p>
          </div>
        </a>
        <p>
          Error: 
          Error when requesting treeherder: Treeherder request error
        </p>
        <p>
          More information about this error has been written to the Web Console.
        </p>
      </section>
    </div>
  </div>
</body>
`;

exports[`App CompareResults or CompareOverTime loader Should render an error page when the treeherder request fails with an error 500 1`] = `
<body>
  <div>
    <div
      aria-live="assertive"
      class="alert-container"
      role="alert"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard fld8l4l css-1qrenvf-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-2fbsan-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        <div
          class="banner-text"
        >
          This is a recently-released version, and we are actively addressing bugs and enhancing performance. If you experience any issues, please
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
            href="https://bugzilla.mozilla.org/enter_bug.cgi?product=Testing&component=PerfCompare&status_whiteboard=[pcf]"
            target="_blank"
          >
            report it on Bugzilla
          </a>
          . 
          Find us at
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
            href="https://matrix.to/#/#perfcompare:mozilla.org"
            target="_blank"
          >
            #perfcompare:mozilla.org on Matrix
          </a>
          .
        </div>
      </div>
    </div>
    <div
      class="fcndgf6"
    >
      <div
        class="MuiGrid-root header-container container_f1ocj467 css-plh9qo-MuiGrid-root"
      >
        <div
          class="toggle-dark-mode f10ie5uy MuiBox-root css-0"
        >
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart toggle-text css-z8chs3-MuiFormControlLabel-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium toggle-switch toggle-dark-mode css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1n50hym-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    aria-label="Dark mode switch"
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    id="Dark mode"
                    name="toggle-dark-mode"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13qdlub-MuiTypography-root"
              >
                Dark mode
              </span>
            </label>
          </div>
        </div>
        <div
          class="header-text MuiBox-root css-0"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter MuiTypography-gutterBottom perfcompare-header css-160ox9e-MuiTypography-root"
          >
            PerfCompare
          </h1>
        </div>
      </div>
      <section
        class="container_fk3n580"
      >
        <a
          aria-label="link to home"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qhous5-MuiTypography-root-MuiLink-root"
          href="/"
        >
          <div
            class="css-1d9cypr-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1mecf1h-MuiSvgIcon-root"
              data-testid="ChevronLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <p>
              Home
            </p>
          </div>
        </a>
        <p>
          Error: 
          Error when requesting treeherder: (500) Internal Server Error
        </p>
        <p>
          More information about this error has been written to the Web Console.
        </p>
      </section>
    </div>
  </div>
</body>
`;
